packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1170,7 +1170,7 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
       assert(!_usesRouter);
       _defaultRouteInformationProvider?.dispose();
       _defaultRouteInformationProvider = null;
-      if (oldWidget == null || widget.navigatorKey != oldWidget.navigatorKey) {
+      if (_navigator == null || widget.navigatorKey != oldWidget!.navigatorKey) {
```

Comment:
This change doesn't seem to be part of "fix a MaterialApp NNBD issue" and there's no test for it. Maybe you didn't intend to include it?